File: android/TodokTodok/src/main/java/todoktodok/backend/reply/application/dto/response/ReplyResponse.java
package todoktodok.backend.reply.application.dto.response;

import todoktodok.backend.member.application.dto.response.MemberResponse;
import todoktodok.backend.reply.domain.Reply;

import java.time.LocalDateTime;

public record ReplyResponse(
        Long replyId,
        MemberResponse member,
        LocalDateTime createdAt,
        String content,
        int likeCount,
        boolean isLikedByMe
) {

    public ReplyResponse(
            final Reply reply,
            final int likeCount,
            final boolean isLikedByMe
    ) {
        this(
                reply.getId(),
                new MemberResponse(reply.getMember()),
                reply.getCreatedAt(),
                reply.getContent(),
                likeCount,
                isLikedByMe
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/comment/ReplyResponse.kt
package com.team.todoktodok.data.network.response.comment

import com.team.domain.model.Reply
import com.team.todoktodok.data.core.ext.toLocalDateTime
import com.team.todoktodok.data.network.response.discussion.MemberResponse
import com.team.todoktodok.data.network.response.discussion.toDomain
import kotlinx.serialization.Serializable

@Serializable
data class ReplyResponse(
    val content: String,
    val createdAt: String,
    val isLikedByMe: Boolean,
    val likeCount: Int,
    val member: MemberResponse,
    val replyId: Int,
)

fun ReplyResponse.toDomain() =
    Reply(
        content,
        createdAt.toLocalDateTime(),
        likeCount,
        member.toDomain(),
        replyId,
        isLikedByMe,
    )
